src/Uri.php

```diff
@@ -301,12 +301,17 @@ public function parse($uri)
                 $this->setUserInfo($userInfo);
             }
 
-            $nMatches = preg_match('/:[\d]{1,5}$/', $authority, $matches);
+            $nMatches = preg_match('/:[\d]{0,5}$/', $authority, $matches);
             if ($nMatches === 1) {
                 $portLength = strlen($matches[0]);
                 $port = substr($matches[0], 1);
 
-                $this->setPort((int) $port);
+                // If authority ends with colon, port will be empty string.
+                // Remove the colon from authority, but keeps port null
+                if ($port && $port !== '') {
```

Comment:
`if ($port)` is sufficient here

Reply:
I know, but I preferred keeping the strict compare. I'll change it right now